Microsoft.Bot.Builder.LUIS.Tests/LuisRecognizerTests.cs

```diff
@@ -0,0 +1,170 @@
+            var result = await luisRecognizer.Recognize("Please deliver it to 98033 WA", CancellationToken.None);
+            Assert.IsNotNull(result);
+            Assert.IsNotNull(result.Text);
+            Assert.AreEqual("Please deliver it to 98033 WA", result.Text);
+            Assert.IsNotNull(result.Intents);
+            Assert.IsNotNull(result.Intents["Delivery"]);
+            Assert.IsNotNull(result.Entities);
+            Assert.IsNull(result.Entities["builtin_number"]);
+            Assert.IsNull(result.Entities["State"]);
+            Assert.IsNotNull(result.Entities["Address"]);
+            Assert.AreEqual(98033, result.Entities["Address"][0]["builtin_number"][0]);
+            Assert.AreEqual("wa", result.Entities["Address"][0]["State"][0]);
+            Assert.IsNotNull(result.Entities["$instance"]);
+            Assert.IsNull(result.Entities["$instance"]["builtin_number"]);
+            Assert.IsNull(result.Entities["$instance"]["State"]);
+            Assert.IsNotNull(result.Entities["$instance"]["Address"]);
+            Assert.AreEqual(21, result.Entities["$instance"]["Address"][0]["startIndex"]);
+            Assert.AreEqual(28, result.Entities["$instance"]["Address"][0]["endIndex"]);
+            Assert.IsTrue((double)result.Entities["$instance"]["Address"][0]["score"] >= 0 && (double)result.Entities["$instance"]["Address"][0]["score"] <= 1);
+            Assert.IsNotNull(result.Entities["Address"][0]["$instance"]);
+            Assert.IsNotNull(result.Entities["Address"][0]["$instance"]["builtin_number"]);
+            Assert.AreEqual(21, result.Entities["Address"][0]["$instance"]["builtin_number"][0]["startIndex"]);
+            Assert.AreEqual(25, result.Entities["Address"][0]["$instance"]["builtin_number"][0]["endIndex"]);
+            Assert.AreEqual("98033", result.Entities["Address"][0]["$instance"]["builtin_number"][0]["text"]);
+            Assert.IsNotNull(result.Entities["Address"][0]["$instance"]["State"]);
+            Assert.AreEqual(27, result.Entities["Address"][0]["$instance"]["State"][0]["startIndex"]);
+            Assert.AreEqual(28, result.Entities["Address"][0]["$instance"]["State"][0]["endIndex"]);
+            Assert.AreEqual("wa", result.Entities["Address"][0]["$instance"]["State"][0]["text"]);
+            Assert.IsTrue((double)result.Entities["Address"][0]["$instance"]["State"][0]["score"] >= 0 && (double)result.Entities["Address"][0]["$instance"]["State"][0]["score"] <= 1);
+        }
+
+        private static IRecognizer GetLuisRecognizer(bool verbose = false, ILuisOptions luisOptions = null)
+        {
+            var luisRecognizerOptions = new LuisRecognizerOptions {Verbose = verbose};
+            var luisModel = new LuisModel("6209a76f-e836-413b-ba92-a5772d1b2087", "f2eef6a1cab345b9b4b53743357e869f", new Uri("https://westus.api.cognitive.microsoft.com/luis/v2.0/apps/"), LuisApiVersion.V2);
```

Comment:
This... isn't allowed. No checking keys into github. 

You'll now have to depriate / rotate the password. Security teams continually sweep for this stuff (even in GitHub history and branches) and will send you a nasty mail.

Reply:
Why is the URI tied to westus? Wouldn't the DNS name that's one of the Anycast / GeoDNS names be better?

Reply:
Moving these values to environment variables (similar to how other recognizers are being tested). 
LUIS doesn't have a global DNS (something to do with how keys are provisioned per region), specific region must be specified.